### PR TITLE
shap 0.46.0

### PR DESCRIPTION
**Pull request by lorepirri**
shap 0.46.0

**Destination channel:** defaults

### Links

- [PKG-5654]
- dev_url (pypi): https://github.com/shap/shap/tree/v0.46.0
- diff: https://github.com/shap/shap/compare/v0.45.1...v0.46.0
- conda-forge: https://github.com/conda-forge/shap-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/shap/0.46.0
- pypi inspector: https://inspector.pypi.io/project/shap/0.46.0

### Explanation of changes:

- bump to 0.46.0
- add upstream tests (not all the test dependencies are available, but are optional)
- delete `recipe/yum_requirements.txt`

[PKG-5654]: https://anaconda.atlassian.net/browse/PKG-5654?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lorepirri** [comment]
@AnacondaRecipes/psubs this is read for review.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
-  skip: true # [py<37 or s390x]
+  # s390x: numba not available
+  skip: true  # [py<39 or s390x]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - make  # [unix]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: I don't think make is needed. It's using pure setuptools stuff: https://github.com/shap/shap/blob/v0.46.0/setup.py#L101-L111

Comment (lorepirri) [reply recipe/meta.yaml]: thanks, that is right. Removed.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -68,14 +101,17 @@ about:
...
     SHAP (SHapley Additive exPlanations) is a unified approach to explain the output
-    of any machine learning model. SHAP connects game theory with local explanations, 
-    uniting several previous methods and representing the only possible consistent 
-    and locally accurate additive feature attribution method based on expectations.
-  doc_url: https://shap.readthedocs.io/en/latest/
+    of any machine learning model. SHAP connects game theory with local explanations,
+    uniting several previous methods and representing the only possible consistent and
+    locally accurate additive feature attribution method based on expectations.
+  doc_url: https://shap.readthedocs.io
   dev_url: https://github.com/slundberg/shap
```
Comment (ifitchet) [inline recipe/meta.yaml]:
```suggestion
  dev_url: https://github.com/shap/shap
```

**Review by ifitchet — COMMENTED** [review]
The old one redirects

**Comment by lorepirri** [comment]
> The old one redirects

Nice catch, thanks!

**Review by ifitchet — APPROVED** [review]
LGTM

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**